### PR TITLE
Use Map.of/List.of/Set.of over Collections and Arrays.asList

### PR DESCRIPTION
**Pull request by DanVanAtta**
As discussed in: https://github.com/triplea-game/triplea/issues/5524

Favor using List.of instead Collections.emptyList, Collections.singletonList,
and Arrays.asList.

 

## Functional Changes

[] New map or map update
[] New Feature
[] Feature update or enhancement
[] Feature Removal
[x] Code Cleanup or refactor
[] Configuration Change
[] Bug fix:  
[] Other:   

## Testing

[x] Manually testing done

- Ran through 2 AI vs AI games 

## Additional Review Notes

- Not entirely safe as `Arrays.asList` is mutable.  The manual testing hopefully found everywhere we would have a problem. I've some concern we may see issues in the file menus, with luck we'll find those, if any, with further prerelease testing
- Not 100% of usagse of emtpyList/emptySet were converted, some tests relied on being able to add `null`, in those cases those tests were left alone.

### CODE REVIEW

**Comment by asvitkine** [comment]
Is it possible to wait on this until my move refactor pr is landed? Presumably this change should be easy to rebase since it's just a search and replace and spotless apply.

Also one case to be careful of is that Collections.singletonList(null) is allowed but not List.of(null).

**Comment by DanVanAtta** [comment]
@asvitkine we try to review/merge in order to add predictability to conflicts. With that said, we have not really tested that policy with WIP PRs. If having to choose, I would generally pick the PR that touches more files. While you may see conflicts in a dozen lines, this PR is very likely to see conflicts in hundreds of lines. If we try a package move while this update is in-flight, the conflicts will be pretty killer.

This update was a bit more than just search and replace, missing imports had to be fixed by hand. We had plenty examples where the `List` import was not present, eg: `Collection<> someCollection = Collections.emptyList()`

> Also one case to be careful of is that Collections.singletonList(null) is allowed but not List.of(null).

Indeed, we had one example of that. Using null values in lists is pretty archaic, happily there was only the one example to fix. The update from `Arrays.asList` to `List.of` was a big concern as we're going from mutable to immutable. Note though, I had the AI play against itself for about an hour before cutting this PR.

> Is it possible to wait on this until my move refactor pr is landed?

I checked for merge conflicts, it does not conflict.

**Comment by asvitkine** [comment]
> @asvitkine we try to review/merge in order to add predictability to conflicts. With that said, we have not really tested that policy with WIP PRs. If having to choose, I would generally pick the PR that touches more files. While you may see conflicts in a dozen lines, this PR is very likely to see conflicts in hundreds of lines. If we try a package move while this update is in-flight, the conflicts will be pretty killer.
> 
> This update was a bit more than just search and replace, missing imports had to be fixed by hand. We had plenty examples where the `List` import was not present, eg: `Collection<> someCollection = Collections.emptyList()`
> 
> > Also one case to be careful of is that Collections.singletonList(null) is allowed but not List.of(null).
> 
> Indeed, we had one example of that. Using null values in lists is pretty archaic, happily there was only the one example to fix. The update from `Arrays.asList` to `List.of` was a big concern as we're going from mutable to immutable. Note though, I had the AI play against itself for about an hour before cutting this PR.
> 
> > Is it possible to wait on this until my move refactor pr is landed?
> 
> I checked for merge conflicts, it does not conflict. 👍

Okay, sounds good. Thanks for checking!

**Comment by DanVanAtta** [comment]
@asvitkine thanks for taking the look! I'm not sure how to handle your feedback. There are 500+ locations changed, I don't think it's reasonable for us to fix up every one and them and have full confidence nothing is wrong. TripleA is super brittle, the chance of this being reverted if we do that would be pretty high, and then it would be for nought. Even if we did make all updates we could, the scope could increase by quite a bit (starts us down a rabbit hole), and/or we might wind up spending a lot more time testing.

We could add TODO statements to places where we maybe a collection copy is not needed, but I would be concerned we'd add them in locations where we're not sure and are speculating. I think overall we should try to follow-up with further clean-ups. The existing scope was really to remove the empty lists, I already spent 2 and half hours on this update, the ROI of spending another two hours is kinda poor. 

Do you have any thoughts or suggestions of good ways to make sure we don't lose the review feedback and successfully follow-up?

**Comment by asvitkine** [comment]
> @asvitkine thanks for taking the look! I'm not sure how to handle your feedback. There are 500+ locations changed, I don't think it's reasonable for us to fix up every one and them and have full confidence nothing is wrong. TripleA is super brittle, the chance of this being reverted if we do that would be pretty high, and then it would be for nought. Even if we did make all updates we could, the scope could increase by quite a bit (starts us down a rabbit hole), and/or we might wind up spending a lot more time testing.
> 
> We could add TODO statements to places where we maybe a collection copy is not needed, but I would be concerned we'd add them in locations where we're not sure and are speculating. I think overall we should try to follow-up with further clean-ups. The existing scope was really to remove the empty lists, I already spent 2 and half hours on this update, the ROI of spending another two hours is more than I'm willing to invest.
> 
> Do you have any thoughts or suggestions of good ways to make sure we don't lose the review feedback and successfully follow-up?

That's fair. I'm fine if we want to do it in a follow up. I can send a PR if you want after you land this.

File: game-core/src/test/java/games/strategy/triplea/delegate/WW2V3Year41Test.java [hunk 1]
```diff
@@ -1166,7 +1164,7 @@ void testAttackDestroyerAndSubsAgainstSubAndDestroyer() {
         invocation -> {
           final Collection<Unit> selectFrom = invocation.getArgument(0);
           return new CasualtyDetails(
-              Collections.singletonList(selectFrom.iterator().next()), new ArrayList<>(), false);
+              List.of(selectFrom.iterator().next()), new ArrayList<>(), false);
```
Comment (asvitkine) [inline game-core/src/test/java/games/strategy/triplea/delegate/WW2V3Year41Test.java]: List.of() can be used for the next param as well. I checked CasualtyDetails and it makes a copy.

**Comment by DanVanAtta** [comment]
> That's fair. I'm fine if we want to do it in a follow up. I can send a PR if you want after you land this.

That would be really great @asvitkine! I can certainly help review it

**Pull request merged**